src/server/game/Chat/Channels/Channel.cpp

```diff
@@ -704,6 +704,11 @@ void Channel::Say(ObjectGuid const& guid, std::string const& what, uint32 lang,
     if (sWorld->getBoolConfig(CONFIG_ALLOW_TWO_SIDE_INTERACTION_CHANNEL))
         lang = LANG_UNIVERSAL;
 
+    Player* player = ObjectAccessor::FindConnectedPlayer(guid);
+
+    if (!player->GetSession()->IsAddonRegistered(prefix))
```

Comment:
I don't think that this condition should be here. The message to be sent might not be related to addons at all.